backend/src/services/v2/model.ts

```diff
@@ -147,6 +147,18 @@ export async function getModelCardRevision(user: UserDoc, modelId: string, versi
   return modelCard
 }
 
+export async function getModelCardRevisions(user: UserDoc, modelId: string) {
+  const modelCardRevisions = await ModelCardRevisionModel.find({ modelId })
+
+  if (!modelCardRevisions) {
```

Comment:
I think this needs revisiting. Currently `ModelCardRevisionModel.find()` will just return an empty array, which means it shouldn't ever be undefined. 

I would probably do an additional check before this line to make sure that the `modelId` returns an actual model object. Then you can do your undefined error check on that.